internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,29 +116,148 @@ func TestDelayConsumer_Run(t *testing.T) {
 	dc.Run(ctx)
 }
 
-func getDummyBrokerMessage() messagebroker.ReceivedMessage {
-	tNow := time.Now()
-	tPast := tNow.Add(time.Second * 100 * -1)
+func TestDelayConsumer_Run_Consume(t *testing.T) {
+	ctx, ctrl, subCtx, cancel, brokerStore, handler, consumer, cache := setupDC(t)
+	subs := getMockSubscriptionModel()
+	subscriberID := "subscriber-id"
+	errCh := make(chan error)
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1", subs, brokerStore, handler, cache, errCh)
+	producer := getMockProducer(ctx, ctrl)
+	brokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+
+	messages := make([]messagebroker.ReceivedMessage, 0)
+	// message with delay within duration of test-case
+	msg1 := getDummyBrokerMessage("msg-1", "msg-id-1", time.Now().Add(time.Nanosecond*1000), 5)
+	// message with delay outside duration of test-case
+	msg2 := getDummyBrokerMessage("msg-2", "msg-id-2", time.Now().Add(time.Second*2), 5)
```

Comment:
These two should be two independent test cases to verify both conditions. A failure here would mean we debug the entire test rather than individual test cases.

Reply:
Splited into two test case